panos/network.py

```diff
@@ -2125,6 +2126,190 @@ def _setup(self):
+        interval_seconds (int): Interval Seconds (sec)
+        update_intervals (int): Update Intervals
+    """
+
+    NAME = None
+    CHILDTYPES = (
+        "network.RipInterface",
+        "network.RipAuthProfile",
+        "network.RipExportRules",
+    )
+
+    def _setup(self):
+        # xpaths
+        self._xpaths.add_profile(value="/protocol/rip")
+
+        # params
+        params = []
+
+        params.append(
+            VersionedParamPath("enable", path="enable", default=True, vartype="yesno")
+        )
+        params.append(
+            VersionedParamPath(
+                "reject_default_route",
+                default=True,
+                vartype="yesno",
+            )
+        )
+        params.append(
+            VersionedParamPath(
+                "allow_redist_default_route",
+                path="allow-redist-default-route",
+                vartype="yesno",
+            )
+        )
+        params.append(
+            VersionedParamPath(
+                "delete_intervals", path="/timers/delete-intervals", vartype="int"
+            )
+        )
+        params.append(
+            VersionedParamPath(
+                "expire_intervals", path="/timers/expire-intervals", vartype="int"
+            )
+        )
+        params.append(
+            VersionedParamPath(
+                "interval_seconds", path="/timers/interval-seconds", vartype="int"
+            )
+        )
+        params.append(
+            VersionedParamPath(
+                "update_intervals", path="/timers/update-intervals", vartype="int"
+            )
+        )
+
+        self._params = tuple(params)
+
+
+class RipInterface(VersionedPanObject):
+    """Rip Interface
+
+    Add to a :class:`panos.network.Rip` instance.
+
+    Args:
+        name (str): Interface name
+        enable (bool): Enable
+        advertise_default_route_metric (int): default route metric, enables advertise_default_route
+        auth_profile (str): Auth profile name
+        mode (str): Any of "normal", "passive", or "send-only"
+    """
+
+    SUFFIX = ENTRY
+
+    def _setup(self):
+        self._xpaths.add_profile(value="/interface")
+
+        params = []
+
+        params.append(VersionedParamPath("enable", path="enable", vartype="yesno"))
+        params.append(
+            VersionedParamPath(
+                "advertise_default_route_metric",
+                path="default-route/advertise/metric",
+                vartype="int",
+            )
+        )
+        params.append(VersionedParamPath("auth_profile", path="authentication"))
+        params.append(
+            VersionedParamPath(
+                "mode", path="mode", values=["normal", "passive", "send-only"]
+            )
+        )
+
+        self._params = tuple(params)
+
+
+class RipAuthProfile(VersionedPanObject):
+    """Rip Authentication Profile
+
+    Args:
+        name (str): Name of Auth Profile
+        type (str): 'password' or 'md5'
```

Comment:
Please rename this param (e.g. - `auth_type`).  Naming variables `type` is trouble.

Reply:
Renamed param